network-view-docs: rename source file (#3183)
Unleash/unleash: website/docs/reference/network-view.mdx

---
title: Network view
---

import Figure from '@site/src/components/Figure/Figure.tsx'

:::info Availability

The network view was released in Unleash 4.21. It is available to Pro and Enterprise users.

You must configure the [data source](#data-source) to activate the feature.

:::

The Unleash admin UI contains a **network view** as part of the admin configuration pages. This network view was designed to give you an overview and understanding of incoming requests to your Unleash instance. It makes it possible to pinpoint sources of sudden request surges, and can therefore also help you identify issues with SDK setups[^1].

The network view offers two different visualizations of the same data, known as the [network overview](#network-overview) and the [network traffic](#network-traffic) views.

> Because Unleash doesn't store this kind of data itself, the network view requires you to configure an [external data source](#data-source) for the network overview to work. The network view is only mad available if you tell Unleash where it can find the data (refer to the [data source section](#data-source)).

Reviewer: ```suggestion
Because Unleash doesn't store this kind of data itself, the network view requires you to configure an [external data source](#data-source) for the network overview to work. The network view is only available if you tell Unleash where it can find the data (refer to the [data source section](#data-source)).
```
I think this was "made available", but also it can just be removed

Author: Absolutely! Good catch.

The network view is intended to provide a simple and Unleash-centric overview that serves basic use cases. If you need detailed metrics and connection graphs, you may be better off using specialized network monitoring tools.

## Applications

> Both the network overview and the network traffic diagrams show you **applications** that have made requests to the Unleash instance. An **application** is defined as anything that sends requests to [Unleash client API](/reference/api/unleash/client), such as [Unleash SDKs](./sdks/index.md), [Unleash Edge](/docs/generated/unleash-edge.md), or the [Unleash proxy](/docs/generated/unleash-proxy.md).

Reviewer: ```suggestion
Both the network overview and the network traffic diagrams show you **applications** that have made requests to the Unleash instance. An **application** is defined as anything that sends requests to [Unleash client API](/reference/api/unleash/client), such as [Unleash SDKs](./sdks/index.md), [Unleash Edge](/docs/generated/unleash-edge.md), or the [Unleash proxy](/docs/generated/unleash-proxy.md), even the admin UI.
```
Admin UI is one of those cases of unknown applications because it doesn't really matter much, but we wanted to have it anyway

Author: Got it! In that case, how about this? Because it's anything that sends something to `/api/admin` too, right?

```suggestion
Both the network overview and the network traffic diagrams show you **applications** that have made requests to the Unleash instance. An **application** is defined as anything that sends requests to the [Unleash client API](/reference/api/unleash/client) or the [Unleash admin API](/reference/api/unleash/admin). This includes as [Unleash SDKs](./sdks/index.md), [Unleash Edge](/docs/generated/unleash-edge.md), the [Unleash proxy](/docs/generated/unleash-proxy.md), and even the admin UI.
```

Reviewer: Anything that sends requests to `/api/` whether it exist now or anything we create in the future. So `/api/frontend` too

Author: Got it! I think this clarifies that a little better.

```suggestion
Both the network overview and the network traffic diagrams show you **applications** that have made requests to the Unleash instance. An **application** is defined as anything that sends requests to the [Unleash client API](/reference/api/unleash/client), the [Unleash admin API](/reference/api/unleash/admin), the [Unleash front-end API](front-end-api.md), or any other API that Unleash exposes. This includes [Unleash SDKs](./sdks/index.md), [Unleash Edge](/docs/generated/unleash-edge.md), the [Unleash proxy](/docs/generated/unleash-proxy.md), and even the admin UI.
```

### "unknown" applications

Requests that come from Unleash SDKs and other official Unleash applications will always have an application name defined. But you might sometimes see some applications listed as "unknown" in the diagrams.

> This happens when Unleash receives requests that don't contain an application name. This can happen, for instance, if you make HTTP requests from the command line to test Unleash connections or if you write your own HTTP client for Unleash that doesn't provide an application name.

Reviewer: ```suggestion
This happens when Unleash receives requests that don't contain an application name header. This can happen, for instance, if you make HTTP requests from the command line to test Unleash connections or if you write your own HTTP client for Unleash that doesn't provide an application name.
```
I couldn't find a lot of documentation about this header, it appears in this legacy page: https://docs.getunleash.io/reference/api/legacy/unleash/client/features#fetching-feature-toggles and nowhere else... most SDKs will set it up automatically, but it's not documented. So perhaps we can add it here, WDYT?

Author: Oh, interesting! I didn't know, to be honest. I think that should be in the OpenAPI docs too (but this isn't the place for those changes). It should be documented somewhere, but I don't want to hold this back anymore if we can help it. How about this?

```suggestion
This happens when Unleash receives requests that don't contain an application name header (`UNLEASH_APPNAME`). This can happen, for instance, if you make HTTP requests from the command line to test Unleash connections or if you write your own HTTP client for Unleash that doesn't provide an application name.
```

Reviewer: That works for me :)
